platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/command/ProfileInstallCommand.java

```diff
@@ -296,13 +278,48 @@ private void stopBundles(BundleService bundleService, List<String> stopBundleNam
     return Optional.ofNullable(profileMap);
   }
 
-  private <T> T executeAsSystem(Callable<T> func) {
-    Subject systemSubject = security.getSystemSubject();
-    LOGGER.debug("System Subject retrieved: " + SubjectUtils.getName(systemSubject));
-    if (systemSubject == null) {
-      printError(SECURITY_ERROR);
-      throw new IllegalStateException(SECURITY_ERROR);
+  private void startApplication(ApplicationService applicationService, String application)
+      throws ApplicationServiceException {
+    try {
+      applicationService.startApplication(application);
+      startedAppCount += 1;
+    } catch (ApplicationServiceException e) {
+      printItemStatusFailure("Start Failed: ", application);
+      throw e;
+    }
+  }
+
+  private void installFeature(FeaturesService featuresService, String feature) throws Exception {
+    try {
+      featuresService.installFeature(feature, NO_AUTO_REFRESH);
+      installedFeatureCount += 1;
+    } catch (Exception e) {
+      printItemStatusFailure("Install Failed: ", feature);
+      throw e;
+    }
+  }
+
+  private void uninstallFeature(FeaturesService featuresService, Feature feature) throws Exception {
+    try {
+      featuresService.uninstallFeature(feature.getName(), feature.getVersion(), NO_AUTO_REFRESH);
+      uninstalledFeatureCount += 1;
+    } catch (Exception e) {
+      printItemStatusFailure("Uninstall Failed: ", feature.getName());
+      throw e;
+    }
+  }
+
+  private void stopBundle(BundleService bundleService, String bundle) throws BundleException {
+    try {
+      bundleService.getBundle(bundle).stop();
+      stoppedBundleCount += 1;
+    } catch (BundleException e) {
+      printItemStatusFailure("Stop Failed: ", bundle);
+      throw e;
+    } catch (IllegalArgumentException ie) {
+      printItemStatusFailure("Stop Failed: ", bundle);
+      printError(String.format("Bundle: %s does not exist", bundle));
```

Comment:
❓Would an `IllegalArgumentException` ever be thrown for these other reasons https://github.com/apache/karaf/blob/karaf-4.1.2/bundle/core/src/main/java/org/apache/karaf/bundle/core/internal/BundleServiceImpl.java#L158-L174? Maybe this message could use the exception message instead.

Reply:
I think it is possible, I can definitely change it to printing the exception. I think when I originally wrote this I was trying to avoid printing exceptions directly to the user but I can't remember now